src/bblocks/datacommons_tools/custom_data/data_management.py

```diff
@@ -869,26 +898,85 @@ def validate_config(self) -> CustomDataManager:
         self._config.validate_config()
         return self
 
-    def __repr__(self) -> str:
-        input_files_count = len(self._config.inputFiles)
-        sources_count = len(self._config.sources)
-        nodes_count = sum(
-            [len(var) for var in [n.nodes for n in self._mcf_nodes.values()] if var]
-        )
+    def merge_config(
+        self,
+        config: Config | dict | str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+    ) -> CustomDataManager:
+        """Merge ``config`` into the current configuration.
 
-        variables_count = len(self._config.variables or {}) + nodes_count
-        dataframes_count = len(self._data)
+        Args:
+            config: The config to merge. This can be a Config object, a dictionary,
+                or a path to a JSON file.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
 
-        include_input_subdirs = self._config.includeInputSubdirs
-        group_statvars = self._config.groupStatVarsByProperty
+        """
 
-        return (
-            f"<CustomDataManager config: "
-            f"\n{input_files_count} inputFiles, with {dataframes_count} containing data"
-            f"\n{sources_count} sources"
-            f"\n{variables_count} variables"
-            f"\nflags: includeInputSubdirs={include_input_subdirs}, "
-            f"groupStatVarsByProperty={group_statvars}>"
+        if isinstance(config, (str, PathLike)):
+            cfg = Config.from_json(str(config))
+        elif isinstance(config, dict):
+            cfg = Config.model_validate(config)
+        else:
+            cfg = config
+
+        merge_configs(existing=self._config, new=cfg, policy=policy)
+        return self
+
+    def merge_configs_from_directory(
+        self,
+        directory: str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+        replace_loaded_config: bool = True,
+    ) -> CustomDataManager:
+        """Merge all config files found under ``directory``. This will recursively
+        search for config files in subdirectories. It will merge them with whatever
+        config is already loaded in the manager.
+
+        Args:
+            directory: The directory to search for config files.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
+
+        """
+
+        directory_configs = merge_configs_from_directory(
+            directory=directory, policy=policy
         )
 
-    # TODO: add_config method to add a config to the object either from json file, dictionary or another Config object and merge with existing config
+        if replace_loaded_config:
+            self._config = directory_configs
+        else:
+            self.merge_config(directory_configs, policy=policy)
+
+        return self
+
+    @classmethod
+    def from_config_files_in_directory(
```

Comment:
Should this class method just be `from_directory` where it picks up the config files as well as the mfc files?

Reply:
Yeah I suppose that would be a good idea, but perhaps as a separate method (this being slightly more specific to just config files). There may be cases in which people may not want to load massive mcf files that never change but which are stored in the same bucket